catalog/dags/maintenance/add_license_url.py

```diff
@@ -54,148 +55,106 @@ def run_sql(
+        query_id = query_id.replace(char_to_remove, "_")
+
+    conf = {
+        "query_id": query_id,
+        "table_name": "image",
+        "select_query": (
+            f"WHERE license = '{license_}' AND license_version = '{license_version}' "
+            f"AND meta_data->>'license_url' IS NULL"
+        ),
+        # Merge existing metadata with the new license_url
+        "update_query": f"SET meta_data = ({Json(license_url_dict)}::jsonb || meta_data), updated_on = now()",
+        "update_timeout": 259200,  # 3 days in seconds
+        "dry_run": False,
+        "resume_update": False,
+    }
+    return conf
 
-    # Merge existing metadata with the new license_url
-    update_query = dedent(
-        f"""
-        UPDATE image
-        SET meta_data = ({Json(license_url_dict)}::jsonb || meta_data), updated_on = now()
-        WHERE identifier IN (
-            SELECT identifier
-            FROM image
-            WHERE license = '{license_}' AND license_version = '{version}'
-                AND meta_data->>'license_url' IS NULL
-            LIMIT {batch_size}
-            FOR UPDATE SKIP LOCKED
-        );
-        """
-    )
-    total_updated = 0
-    updated_count = 1
-    while updated_count:
-        updated_count = run_sql(
-            update_query,
-            log_sql=total_updated == 0,
-            method="run",
-            handler=RETURN_ROW_COUNT,
-            autocommit=True,
-            dag_task=ti.task,
-        )
-        total_updated += updated_count
-    logger.info(f"Updated {total_updated} rows with {license_url}.")
-
-    return total_updated
-
-
-@task(trigger_rule=TriggerRule.ALL_DONE)
-def report_completion(updated, query: str, ti=None):
-    """
-    Check for null in `meta_data` and send a message to Slack with the statistics
-    of the DAG run.
-
-    :param updated: total number of records updated
-    :param query: SQL query to get the count of records left with `license_url` as NULL
-    :param ti: automatically passed by Airflow, used to set the execution timeout.
-    """
-    total_updated = sum(updated) if updated else 0
-
-    license_groups = run_sql(query, dag_task=ti.task)
-    total_nulls = sum(group[2] for group in license_groups)
-    licenses_detailed = "\n".join(
-        f"{group[0]} \t{group[1]} \t{group[2]}" for group in license_groups
-    )
-
-    message = f"""
-    `{DAG_ID}` DAG run completed. Updated {total_updated} record(s) with `license_url` in the
-    `meta_data` field. Found {len(license_groups)} license groups with {total_nulls} record(s) left pending.
-    """
-    if total_nulls != 0:
-        message += f"\nCount per license-version:\n{licenses_detailed}"
-
-    slack.send_message(
-        message,
-        username="Airflow DAG Data Normalization - license_url",
-        dag_id=DAG_ID,
-    )
 
+@task
+def get_confs(licenses, batch_size: int) -> list[dict]:
+    if not licenses:
+        raise AirflowSkipException("No config required.")
```

Comment:
This sounds confusing. What does "No config required." mean here? Should it be the opposite, "License config required."?

Reply:
Definitely clearer :) Thank you!